### DISCUSSION TITLE
How to make the Blog as the landing page?

### DISCUSSION THREAD

**Discussion by schantanu** [imfing/hextra#354] [Q&A]
First of all, thanks for the great theme.

I want to recreate the blog as the landing page with the Introduction at the top, similar to the following [Home-Info](https://adityatelange.github.io/hugo-PaperMod/posts/papermod/papermod-features/#home-info-mode) mode: 
https://adityatelange.github.io/hugo-PaperMod/

What is the best way to achieve this?

**Comment by imfing** [comment]
Hi, to achieve this, you need to override the index page template in Hextra. See https://imfing.github.io/hextra/docs/advanced/customization/#custom-layouts

Specifically put `layouts/index.html` under your own site, and you may use https://github.com/imfing/hextra/blob/main/layouts/blog/list.html as reference. 

In the future we will provide some utility shortcodes to make this much easier

**Reply by justinryanc** [reply]
I have tried this but it doesn't show the correct content in the blog folder when I do this.  It turns the pages into lists instead of the actual blog posts.  Any idea how to accomplish this?  I tried moving the posts in the root and created a folder for the pages but adds the pages folder to the list as well.

**Reply by moritz1000** [reply]
I'd also love to use such feature. I like Hextra but I only "need" the Blog part, so setting the Blog as the landing page would be awesome.

Same as @justinryanc, I can't the proposed solution working on my end.

**Reply by justinryanc** [reply]
Hey @moritz1000, I was able to get it working on my site.  Take a look at https://github.com/justinryanc/justinryanc.com-v3/blob/main/layouts/home.html for an example of what I did.  It seems to work for now at least, hopefully they add a feature to the theme in the future to fix it.

Here's the part I added, with a bit of help from copilot.

```
{{/* Initialize $pages to an empty collection before the conditional blocks */}}
{{- $pages := slice -}}

{{- $allPosts := where site.RegularPages "Section" "posts" -}}
{{- $sortBy := site.Params.blog.list.sortBy -}}
{{- $sortOrder := site.Params.blog.list.sortOrder -}}

{{- if eq $sortOrder "asc" }}
    {{- $pages = $allPosts.ByParam $sortBy -}}
{{- else }}
    {{- $pages = ($allPosts.ByParam $sortBy).Reverse -}}
{{- end }}

{{/* The $pages variable is now guaranteed to exist here */}}
{{- $pagerSize := site.Params.blog.list.pagerSize | default 10 -}}
{{- $paginator := .Paginate $pages $pagerSize -}}
{{- range $paginator.Pages }}
```

**Reply by moritz1000** [reply]
@justinryanc Thank you so much. I'll have a look at your solution! :)
